### PR TITLE
add autoclick

### PR DESCRIPTION
**Pull request by matthdsm**
depends on #22270

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthdsm** [comment]
@conda-forge/help-python ready for review

File: recipes/autoclick/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - python >3.6,<4.0
+    - poetry >=0.12
+    - poetry-core
+    - pip
+  run:
+    - python >3.6,<4.0
+    - click <7.0,>8.0
```
Comment (matthdsm) [inline recipes/autoclick/meta.yaml:25]:
```suggestion
    - click >7.0,<8.0
```

**Comment by matthdsm** [comment]
@conda-forge/python ready for review

File: recipes/autoclick/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: 583b1c7e57eedd4a9a61b35ff5fbf491d3eb1fedda5895ffaffad7baaf6786d8
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >3.6,<4.0
```
Comment (ocefpaf) [inline recipes/autoclick/meta.yaml:19]:
```suggestion
    - python >=3.6,<4.0
```

File: recipes/autoclick/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - poetry >=0.12
+    - poetry-core
+    - pip
+  run:
+    - python >3.6,<4.0
```
Comment (ocefpaf) [inline recipes/autoclick/meta.yaml:24]:
```suggestion
    - python >=3.6,<4.0
```

File: recipes/autoclick/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    - click >7.0,<8.0
```
Comment (ocefpaf) [inline recipes/autoclick/meta.yaml:25]: pyproject.toml says >=7 but setup.py says ==7 :-/

Let's try this for now.

```suggestion
    - click >=7.0,<8.0
```

File: recipes/autoclick/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - python >=3.6,<4.0
+    - poetry >=0.12
+    - poetry-core
+    - pip
+  run:
+    - python >=3.6,<4.0
+    - click >=7.0,<8.0
```
Comment (ocefpaf) [inline recipes/autoclick/meta.yaml:25]: Looks like we'll need this b/c it is enforced in the setup.py.

```suggestion
    - click ==7.0
```

**Comment by matthdsm** [comment]
Thanks for all the help!

- M

Op 14 mrt. 2023 om 19:48 heeft Filipe ***@***.***> het volgende geschreven:

﻿

Merged #22271<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fconda-forge%2Fstaged-recipes%2Fpull%2F22271&data=05%7C01%7CMatthias.DeSmet%40ugent.be%7C8e9d2a86359947c5833808db24bca875%7Cd7811cdeecef496c8f91a1786241b99c%7C1%7C0%7C638144164912911888%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Wy1us1yvOnZT%2BmrWcKegUmsrS3mTRKzT%2BymB%2BCJpVCw%3D&reserved=0> into main.

**Pull request merged**